src/test/groovy/com/gooddata/md/visualization/VisualizationObjectTest.groovy

```diff
@@ -0,0 +1,220 @@
+        RelativeDateFilter relativeDateFilter = new RelativeDateFilter( new UriObjQualifier("/uri/to/dataSet/2"), "month", 0, -11)
+        VOSimpleMeasureDefinition measureDefinition = new VOSimpleMeasureDefinition(new UriObjQualifier("/uri/to/measure/1"), "sum", false, [positiveAttributeFilter, negativeAttributeFilter, absoluteDateFilter, relativeDateFilter ])
+        JsonNode objectNode = mapper.createObjectNode()
+        objectNode.put("key", "value")
+        objectNode.put("foo", "bar")
+
+        expect:
+        that new VisualizationObject(
+                new VisualizationObject.Content(
+                        new UriObjQualifier("visClass"),
+                        [
+                                new Bucket("bucket1", [attribute1]),
+                                new Bucket("bucket2", [
+                                        new Measure(
+                                                measureDefinition,
+                                                "measure1",
+                                                "Measure 1 alias",
+                                                "Measure 1",
+                                                null
+                                        )
+                                ])
+                        ],
+                        [
+                                positiveAttributeFilter,
+                                negativeAttributeFilter,
+                                absoluteDateFilter,
+                                relativeDateFilter
+                        ],
+                        "{\"key\":\"value\"}",
+                        objectNode
+                ),
+                new Meta("complex")
+        ),
+                jsonEquals(resource(COMPLEX_VISUALIZATION))
+    }
+
+    def "should serialize"() {
+        expect:
+        that new VisualizationObject(
+                new VisualizationObject.Content(
+                        new UriObjQualifier("visClass"),
+                        [],
+                        null,
+                        null,
+                        null
+                ),
+                new Meta("simple")
+        ),
+                jsonEquals(resource(SIMPLE_VISUALIZATION))
+
+    }
+
+    def "should return null when invalid collection requested"() {
+        VisualizationObject stackedChart = readObjectFromResource("/$STACKED_COLUMN_CHART", VisualizationObject)
+        VisualizationObject customChart = readObjectFromResource("/$CUSTOM_CHART", VisualizationObject)
+
+        VisualizationAttribute fromMissingCollection = stackedChart.getAttributeFromCollection(CollectionType.TREND)
+        VisualizationAttribute missingAttribute = customChart.getAttributeFromCollection(CollectionType.VIEW)
+
+        expect:
+        fromMissingCollection == null
+        missingAttribute == null
+    }
+
+    def "return item in view collection"() {
```

Comment:
it looks like the bunch of tests here can be joined to one using `where`, if you need to change the tests names as well, use `@Unroll`

Reply:
with `where` I failed to pass methods getStack, getTrend .. as parameters, is that even possible?